### PR TITLE
Allow hiding progressbar until some time interval has elapsed.

### PR DESCRIPTION
**Pull request by anntzer**
This avoids printing the progressbar for very short loops, i.e. the feature requested in #704.  I am not particularly wedded to the kwarg name, alternatives are welcome.

### CODE REVIEW

**Comment by anntzer** [comment]
Kindly bumping, thanks!

File: tqdm/std.py [hunk 1]
```diff
@@ -775,6 +775,7 @@ def __init__(self, iterable=None, desc=None, total=None, leave=True,
                  unit_scale=False, dynamic_ncols=False, smoothing=0.3,
                  bar_format=None, initial=0, position=None, postfix=None,
                  unit_divisor=1000, write_bytes=None, lock_args=None,
+                 waituntil=0,
```
Comment (casperdcl) [inline tqdm/std.py]: @anntzer how about using `disable=<float>` instead?

We could do:

```python
if disable and disable not in (True, 1):
    disable = False
    waituntil = disable
```

This would be backward-compatible without adding yet another kwarg (similar to `unit_scale`): https://github.com/tqdm/tqdm/blob/80be780ab988c20048ade293ef84d2d8fd9e922a/tqdm/std.py#L380

Comment (anntzer) [reply tqdm/std.py]: It's not clear to me these can be combined -- what if I want to use waituntil=1 for example?

Comment (casperdcl) [reply tqdm/std.py]: Yes as with `unit_scale` it would be treated the same as `True`. You'd need `0.99` or `1.01` etc if you really wanted approx 1 second

Comment (anntzer) [reply tqdm/std.py]: But then I fail to see why you want to combine these into a single kwarg (I'm all for not adding more kwargs than necessary, but not at the cost of combining unrelated stuff)... `disable` and `waituntil` have completely different semantics, and the combination you propose just looks... confusing?
Another way to look at it is, what docstring do you propose?

> disable : bool or float, optional
>
> If True, the entire progressbar wrapper is disabled.
> If None (the default), the progressbar wrapper is disabled on non-TTY.
> If a float value other than 1, the progressbar is not displayed until *disable* seconds have elapsed

?

Comment (casperdcl) [reply tqdm/std.py]: Something like:

```
    disable : bool or float, optional

    [default: False].
    If True or 1, disable the entire progressbar wrapper.
    If None, disabled on non-TTY.
    If any other numeric value, wait until the specified number
    of seconds have elapsed.
```

Comment (anntzer) [reply tqdm/std.py]: Again, I fail to see why you think these two kwargs should be combined.
At the end of the day, though, this is not my call, and I would prefer this feature to go in even if the API may not be the one I like the best, so if you are certain you want to fold the feature into `disable`, then just confirm that and I will change the PR accordingly.

Comment (casperdcl) [reply tqdm/std.py]: I'm not certain about anything :)

Comment (casperdcl) [reply tqdm/std.py]: Perhaps separate kwarg, but rename from `waituntil` to `initinterval` or `startinterval` to be more consistent with existing names?

Comment (anntzer) [reply tqdm/std.py]: Let me rephrase my query, then:  Are willing to merge (or at least review) a PR which introduces the feature under the `disable` kwarg?  If so, then I am willing to ignore my opinion that folding the feature into `disable` is a mistake.

Comment (casperdcl) [reply tqdm/std.py]: I think we're willing to merge this feature, and there are 2 ways to expose it to and end user (new kwarg or augment existing kwarg). Both are easy to implement, but I don't know what users will find best. More opinions from other people would help.

advantages:

- new kwarg
  + no risk of breaking backward compatibility
  + supports exactly 1 second
- augment `disable`
  + no additional kwarg clutter
  + no need to decide on new kwarg name
  + consistency: same implementation approach as `unit_scale`

Comment (anntzer) [reply tqdm/std.py]: OK, I'll let you figure this out then :) (I have already given my arguments above, and don't have much to add.)

Comment (graue70) [reply tqdm/std.py:778]: I'm definitely pro new kwarg. The alternative is WAY confusing. It would make it basically impossible to find the feature at all. If one does, one can use any amount of seconds except 1? Really?

Using `unit_scale` for two things is less bad because nobody wants an actual `unit_scale` of 1.

**Comment by casperdcl** [comment]
fixes #1069

File: tqdm/std.py (hunk 1 above)
Comment (Witiko) [reply tqdm/std.py:778]: I agree with @graue70 and @anntzer. Folding `waituntil` into `disable` seems hostile to the user with respect to discoverability and the unintuitive casting of 1.0 to a boolean.

Comment (Witiko) [reply tqdm/std.py:778]: @casperdcl I agree with @graue70 and @anntzer. Folding `waituntil` into `disable` seems hostile to the user w.r.t. discoverability and the unintuitive casting of 1.0 to a boolean. As for an alternative name, how about `delay`?
